### PR TITLE
[Merged by Bors] - feat: library_search handles iff and Eq.symm

### PR DESCRIPTION
**Pull request by kim-em**
This restores some features from mathlib3's library search, namely automatically finding `Iff.mp`, `Iff.mpr` and `Eq.symm` versions of library lemmas.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/LibrarySearch.lean [hunk 1]
```diff
@@ -41,14 +41,36 @@ private def isBlackListed (declName : Name) : MetaM Bool := do
...
+inductive DeclMod
+| none | symm | mp | mpr
+deriving DecidableEq
+
+initialize librarySearchLemmas : DeclCache (DiscrTree (Name × DeclMod) true) ←
   DeclCache.mk "librarySearch: init cache" {} fun name constInfo lemmas => do
     if constInfo.isUnsafe then return lemmas
     if ← isBlackListed name then return lemmas
-    withNewMCtxDepth do
+    withNewMCtxDepth do withReducible do
```
Comment (dwrensha) [inline Mathlib/Tactic/LibrarySearch.lean:61]: Is this `withReducible` redundant with the other ones below?

Comment (semorrison) [reply Mathlib/Tactic/LibrarySearch.lean:61]: Thanks!

File: test/librarySearch.lean [hunk 2]
```diff
@@ -2,6 +2,7 @@ import Mathlib
 
 noncomputable section
 
+set_option maxHeartbeats 400000 in
```
Comment (fpvandoorn) [inline test/librarySearch.lean:5]: Maybe don't import all of Mathlib so that this test doesn't start failing if Mathlib grows?

**Review by fpvandoorn — COMMENTED** [review]
LGTM

Note that you didn't fix the previous comment yet.

Do we care about applying `Iff.symm` at all?

Is the new behavior much more expensive than the old behavior? Would it make sense to have a cheap(er) version of `library_search` that look as deeply? To be honest, I rarely used `library_search` in Lean 3, because it would always time when importing many files.

bors d+

File: test/librarySearch.lean (hunk 2 above)
Comment (semorrison) [reply test/librarySearch.lean]: Done, good suggestion.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**